MCP Documentation for 26.5 (#44572)
keycloak/keycloak: docs/guides/securing-apps/mcp-authz-server.adoc

Reviewer: Should we perhaps mention that we are planning to add support for resource indicators and are also considering adding support for `OAuth Client ID Metadata Documents`?

Author: I added the following sentences:

>=== For MCP 2025-11-25
>
>==== Client Registration
>
>According to Client Registration Approaches section of the MCP specification, the following three client registration mechanisms are supported and you can choose based on your scenario:
>
>* Client ID Metadata Documents: When client and server have no prior relationship (most common)
>* Pre-registration: When client and server have an existing relationship
>* Dynamic Client Registration: For backwards compatibility or specific requirements
>
>The Keycloak community is planning to suppoort OAuth Client ID Metadata Document in the future. Until the support is completed, {project_name} can still accomodate an MCP client by pre-registration or Dynamic Client Registration.

<#import "/templates/guide.adoc" as tmpl>
<#import "/templates/links.adoc" as links>

<@tmpl.guide

> title="Integrating with MCP (Model Context Protocol)"

Reviewer: ```suggestion
title="Integrating with Model Context Protocol (MCP)"
```

priority=115

> summary="Using {project_name} as an authorization server in MCP">

Author: I fixed the point.

> There are the following four MCP versions:

Reviewer: ```suggestion
There are currently four versions of the Model Context Protocol (MCP) specification:
```

Author: I fixed the point.

* 2024-11-05 (initial version)
* 2025-03-26
* 2025-06-18

> * 2025-11-25 (latest version)

Reviewer: ```suggestion
* 2025-11-25 (latest version)
* 2025-06-18
* 2025-03-26
* 2024-11-05 (initial version)
```

> The initial version is outdated, so this guide treats other three versions. The guide tells you the following:

Reviewer: ```suggestion
The initial version (2024-11-05) does not cover authorization, as such is not covered in this guide. 

This guide tells you the following:
```

Author: I fixed the point.

* Which MCP version {project_name} supports.
* How to set up {project_name} as an authorization server in MCP.

> 

Reviewer: Maybe add a line that says this guide does not cover everything you need to know; and recommend reading the authorization section of the relevant MCP version as well?

> == Standards Compliance MCP requires

Reviewer: Please add `OAuth 2.0 Protected Resource Metadata (RFC 9728)` as MUST, and in Keycloak column add `Requires workaround`

Author: RFC 9728 is for a resource server, namely MCP server. It is not for an authorization server. Therefore I did not add RFC 9728 here.

According to the https://modelcontextprotocol.io/specification/draft/basic/authorization#standards-compliance[MCP specification], there are several standards regarding an authorization server in MCP. The following table shows:

* Which MCP version requires an authorization server to support which standards in which level (MUST, SHOULD, MAY).
* With which standards {project_name} complies.

[%autowidth]
|======
|Standard |2025-03-26 |2025-06-18 |2025-11-25 |{project_name}

| https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-14[Internet Draft - The OAuth 2.1 Authorization Framework]
| MUST
| MUST
| MUST
| Supported

> | https://datatracker.ietf.org/doc/html/rfc8414[RFC 8414 OAuth 2.0 Authorization Server Metadata]

Reviewer: ```suggestion
| https://datatracker.ietf.org/doc/html/rfc8414[OAuth 2.0 Authorization Server Metadata (RFC 8414)]
```

| MUST
| MUST
| MUST
| Supported

> | https://datatracker.ietf.org/doc/html/rfc7591[RFC 7591 OAuth 2.0 Dynamic Client Registration Protocol]

Reviewer: ```suggestion
| https://datatracker.ietf.org/doc/html/rfc7591[OAuth 2.0 Dynamic Client Registration Protocol (RFC 7591)]
```

Author: I fixed the point.

| SHOULD
| SHOULD
| MAY
| Supported

> | https://datatracker.ietf.org/doc/html/draft-ietf-oauth-client-id-metadata-document-00[Internet Draft - OAuth Client ID Metadata Document]

Reviewer: ```suggestion
| https://datatracker.ietf.org/doc/html/draft-ietf-oauth-client-id-metadata-document-00[OAuth Client ID Metadata Document (Internet Draft)]
```
